src/components/elements/typography.ts

```diff
@@ -137,6 +144,149 @@ type HeaderProps = {
+        font_size: '8.0rem',
+        line_height: '8rem',
+        font_size_tabletL: '40px',
+        line_height_tabletL: '50px',
+    },
+    'display-title': {
+        font_size: '6.4rem',
+        line_height: '8rem',
+        font_size_tabletL: '40px',
+        line_height_tabletL: '50px',
+    },
+    'page-title': {
+        font_size: '4.8rem',
+        line_height: '6rem',
+        font_size_tabletL: '32px',
+        line_height_tabletL: '40px',
+    },
+    'section-title': {
+        font_size: '3.2rem',
+        line_height: '4rem',
+        font_size_tabletL: '24px',
+        line_height_tabletL: '30px',
+    },
+    'sub-section-title': {
+        font_size: '2.4rem',
+        line_height: '3rem',
+        font_size_tabletL: '20px',
+        line_height_tabletL: '25px',
+    },
+    'main-paragraph': {
+        font_size: '1.6rem',
+        line_height: '2.4rem',
+        font_size_tabletL: '16px',
+        line_height_tabletL: '24px',
+    },
+    'sub-paragraph': {
+        font_size: '1.4rem',
+        line_height: '2rem',
+        font_size_tabletL: '14px',
+        line_height_tabletL: '21px',
+    },
+    hero: {
+        font_size: '8.0rem',
+        line_height: '10rem',
+        font_size_tabletL: '40px',
+        line_height_tabletL: '50px',
+    },
+    'heading-1': {
+        font_size: '6.4rem',
+        line_height: '8rem',
+        font_size_tabletL: '32px',
+        line_height_tabletL: '40px',
+    },
+    'heading-2': {
+        font_size: '4.8rem',
+        line_height: '6rem',
+        font_size_tabletL: '28px',
+        line_height_tabletL: '34px',
+    },
+    'heading-3': {
+        font_size: '3.2rem',
+        line_height: '4rem',
+        font_size_tabletL: '24px',
+        line_height_tabletL: '30px',
+    },
+    'subtitle-1': {
+        font_size: '2.4rem',
+        line_height: '3.6rem',
+        font_size_tabletL: '18x',
+        line_height_tabletL: '26px',
+    },
+    'subtitle-2': {
+        font_size: '2.0rem',
+        line_height: '3rem',
+        font_size_tabletL: '16px',
+        line_height_tabletL: '24px',
+    },
+    'paragraph-1': {
+        font_size: '1.6rem',
+        line_height: '2.4rem',
+        font_size_tabletL: '14px',
+        line_height_tabletL: '20px',
+    },
+    'paragraph-2': {
+        font_size: '1.4rem',
+        line_height: '2rem',
+        font_size_tabletL: '12px',
+        line_height_tabletL: '18px',
+    },
+    small: {
+        font_size: '1.2rem',
+        line_height: '1.8rem',
+        font_size_tabletL: '10px',
+        line_height_tabletL: '14px',
+    },
+    'extra-small': {
+        font_size: '1.0rem',
+        line_height: '1.4rem',
+        font_size_tabletL: '8px',
+        line_height_tabletL: '12px',
+    },
+}
+export const Typography = styled.h2<HeaderProps>`
+    ${TypographyBaseElement}
+    font-weight: ${({ weight }) => weight || 'bold'};
+    font-size: ${({ size, type }) => {
+        if (size) return size
```

Comment:
as far as i checked size is allowed attribute on DOM elements. dont we have to rename it?

Reply:
Yes, even `type` , But in this PR we are only handling the `colour`. for type and size I guess we can handle it in a parent card that migrate  `Typography` instead of `Header` all over the project.

Reply:
@ali-hosseini-deriv `size`, `type` and `weight` also taken care in this PR, and also making Header component deprecated.